### PR TITLE
calculate 'free_space' based on folder location (external storage)

### PR DESCRIPTION
**Pull request by SergeyMosin**
For example we can have a 50GB google drive and on NC side we have users personal quota of 5GB + 100GB on NFS/NAS external storage ( https://docs.nextcloud.com/server/latest/admin_manual/configuration_files/external_storage_configuration_gui.html ) connected to `/External Storage` folder.

Currently there is no way to import 50GB drive event into the `/External Storage` folder because of '_Not enough space..._` warning.

### CODE REVIEW

**Comment by marcelklehr** [comment]
Hi @SergeyMosin 
Thank you for submitting this pull request! I'm sorry for the delay on our part, this seems like a nice improvement! Could you make sure to sign your commits? You can for example do `git rebase --sign` and then force push this branch.

File: src/components/PersonalSettings.vue [hunk 1]
```diff
@@ -442,7 +442,7 @@ export default {
 					showSuccess(t('integration_google', 'Google options saved'))
 					// callback
 					if (callback) {
-						callback()
```
Comment (marcelklehr) [inline src/components/PersonalSettings.vue:445]: This seems like it would break things. Is this intentional?

Comment (SergeyMosin) [reply src/components/PersonalSettings.vue:445]: The `callback()` function is replaced by `callback(response)`. The `response` var is used on line 836 below and it updates `state.free_space` ( and UI ) to handle the case when a user switches between folders located on different storage back-ends and thus might have different amount of free space.

Comment (marcelklehr) [reply src/components/PersonalSettings.vue:445]: ooh, right. I was confused :see_no_evil: I thought you'd removed the argument :D

**Review by marcelklehr — APPROVED** [review]

**Comment by marcelklehr** [comment]
Thank you for your contribution!

**Pull request merged**